public/signer.js

```diff
@@ -190,44 +203,68 @@ const ethers = require("ethers");
 const axios = require("axios");
 const sign = require("./signer.js").sign;
 
-async function signerMakeTx({
-  privateKey,
+function createTransactionSigner({
   appAddress,
-  rpc,
-  gateway,
+  provider,
   forwarderAddress,
+  gateway,
   accessToken,
-  method,
-  value,
 }) {
-  const wallet = new ethers.Wallet(privateKey);
-  const arcanaContract = new ethers.Contract(appAddress, arcana.abi, wallet);
-  const provider = new ethers.providers.JsonRpcProvider(rpc);
+  const ethProvider = new ethers.providers.Web3Provider(provider);
+  const arcanaContract = new ethers.Contract(
+    appAddress,
+    arcana.abi,
+    ethProvider
+  );
   const forwarderContract = new ethers.Contract(
     forwarderAddress,
     forwarder.abi,
-    provider
+    ethProvider
   );
-  const req = await sign(
-    wallet,
-    arcanaContract,
-    forwarderContract,
-    method,
-    value
-  );
-  const res = await axios.post(gateway + "/meta-tx/", req, {
-    headers: {
-      Authorization: "Bearer " + accessToken,
-    },
-  });
-  await new Promise((r) => setTimeout(r, 1000));
-  const tx = await provider.getTransaction(res.data.txHash);
-  await tx.wait();
-  console.log(method, "Done");
-  return res.data;
+
+  return async function signTransaction(method, value) {
+    const req = await sign(
+      ethProvider,
+      arcanaContract,
+      forwarderContract,
+      method,
+      value
+    );
+    const res = await axios.post(gateway + "/meta-tx/", req, {
+      headers: {
+        Authorization: "Bearer " + accessToken,
+      },
+    });
+    const tx = await ethProvider.getTransaction(res.data.txHash);
+    await tx.wait();
+    return res.data;
+  };
 }
 
-window.signerMakeTx = signerMakeTx;
+function hashJson(data) {
+  return ethers.utils.id(JSON.stringify(data));
+}
+
+async function generateLoginInfo({ provider, gateway }) {
+  const ethProvider = new ethers.providers.Web3Provider(provider);
+  let address = await ethProvider.getSigner().getAddress();
+  let nonce = (await axios.get(gateway + `/get-nonce/?address=${address}`))
```

Comment:
replace with `const`

Reply:
Need to do it in [DashboardSigner](https://github.com/arcana-network/DashboardSigner/blob/544e6849897f6303429ac25f93192225acd4670e/main.js#L51-L53). We bundle this file from there. Noted it and opened a new issue in the repo, will update it later.